lib/upgradable-multisig/multisig/utils.ak

```diff
@@ -0,0 +1,62 @@
+use aiken/list
+use aiken/transaction.{Input, Output}
+use aiken/transaction/credential.{Credential}
+use aiken/transaction/value.{PolicyId, Value}
+use upgradable_multisig/types.{AssetClass, MultisigDatum, PubKeyHash}
+
+pub fn get_input_by_token(inputs: List<Input>, policy_id: PolicyId) {
+  list.find(
+    inputs,
+    fn(input) { list.has(value.policies(input.output.value), policy_id) },
+  )
+}
+
+pub fn validate_multisig_datum(datum: MultisigDatum) -> Bool {
+  (list.length(datum.signers) > 1)? && (datum.threshold > 0)? && (datum.spending_limit > 2_000_000)?
+}
+
+// Helper function to get the amount of a specific asset from a Value
+pub fn get_asset_amount(value: Value, asset: AssetClass) -> Int {
+  value.quantity_of(value, asset.policy_id, asset.asset_name)
+}
+
+// Helper function to validate where the NFT goes
+pub fn validate_token_output(
+  output: Output,
+  policy: PolicyId,
+  token_name: ByteArray,
+  credential: Credential,
+  is_credential: Bool,
```

Comment:
Is this only used with the `True` value? If so, then we should drop the other use case.

Reply:
Yes it is. I've adjusted accordingly.